### PR TITLE
Feature/pressure-plugin

### PR DESCRIPTION
**Pull request by hitimo**
Implemented by @pvechersky; based on https://github.com/pvechersky/rotors_simulator/tree/feature/fixed_wing_sim

### CODE REVIEW

**Comment by pvechersky** [comment]
@hitimo Thank you for opening the pull request for this! I can adapt this to the new ROS/Gazebo interface in a bit

**Comment by pvechersky** [comment]
Adapted the old pressure plugin to work with the new ROS-Gazebo interface.

**Comment by hitimo** [comment]
Thanks @pvechersky !

**Comment by hitimo** [comment]
jenkins@`mc-xenial-kinetic` is offline, that's why all PR's are pending..

**Comment by schneith** [comment]
test this please

**Comment by hitimo** [comment]
test this please

File: rotors_gazebo_plugins/include/rotors_gazebo_plugins/gazebo_pressure_plugin.h [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+namespace gazebo {
+// Constants
+static constexpr double kRs = 8314.32; /* Nm/ (kmol K), gas constant */
+static constexpr double kM0 = 28.9644; /* kg/kmol, mean molecular weight of air */
+static constexpr double kG0 = 9.80665; /* m/s^2, acceleration due to gravity at 45.5425 deg lat */
+static constexpr double kR0 = 6356766.0; /* m, Earth radius at g0 */
+static constexpr double kP0 = 101325.0; /* Pa, air pressure at g0 */
+static constexpr double kT0 = 288.15; /* K, standard sea-level temperature */
+static constexpr double kTl = 0.0065; /* K/m, temperature lapse */
+static constexpr double kAs = kG0 * kM0 / (kRs * -kTl);
```
Comment (ffurrer) [inline rotors_gazebo_plugins/include/rotors_gazebo_plugins/gazebo_pressure_plugin.h]: Can we be a bit more verbose for these variable names.

Comment (hitimo) [inline rotors_gazebo_plugins/include/rotors_gazebo_plugins/gazebo_pressure_plugin.h]: Agree. @pvechersky Maybe just use the comments as variable names (incl. units) and remove comments.

File: rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp [hunk 2]
```diff
@@ -103,13 +103,14 @@ void GazeboPressurePlugin::OnUpdate(const common::UpdateInfo& _info) {
 
   // Compute the geopotential altitude.
   double z = ref_alt_ + model_->GetWorldPose().pos.z;
```
Comment (hitimo) [inline rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp]: `z` --> `height_geopotential_m`?

File: rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp [hunk 3]
```diff
@@ -103,13 +103,14 @@ void GazeboPressurePlugin::OnUpdate(const common::UpdateInfo& _info) {
...
-  double h = kR0 * z / (kR0 + z);
+  double h = kEarthRadiusMeters * z / (kEarthRadiusMeters + z);
```
Comment (hitimo) [inline rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp]: `h` --> `?`

File: rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp [hunk 4]
```diff
@@ -103,13 +103,14 @@ void GazeboPressurePlugin::OnUpdate(const common::UpdateInfo& _info) {
...
 
   // Compute the temperature at the current altitude.
-  double t = kT0 - kTl * h;
+  double t = kSeaLevelTempKelvin - kTempLapseKelvinPerMeter * h;
```
Comment (hitimo) [inline rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp]: `t` --> `temperature_at_altitude_kelvin`

File: rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp [hunk 5]
```diff
@@ -103,13 +103,14 @@ void GazeboPressurePlugin::OnUpdate(const common::UpdateInfo& _info) {
...
 
   // Compute the current air pressure.
-  double p = kP0 * exp(kAs * log(kT0 / t));
+  double p = kPressureOneAtmospherePascals *
```
Comment (hitimo) [inline rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp]: `p` --> `pressure_at_altitude_pascal`

**Comment by hitimo** [comment]
Perfect, thanks @pvechersky

**Comment by hitimo** [comment]
test this please

**Pull request merged**